### PR TITLE
qemu-arm: Add ldscript dependency in the final firmware.elf target.

### PR DESCRIPTION
**Pull request by Jongy**

### CODE REVIEW

File: ports/qemu-arm/Makefile [hunk 1]
```diff
@@ -127,8 +127,8 @@ run: $(BUILD)/firmware.elf
 	qemu-system-arm -machine $(BOARD) $(QEMU_EXTRA) -nographic -monitor null -semihosting -kernel $<
 
 ## `$(LD)` doesn't seem to like `--specs` for some reason, but we can just use `$(CC)` here.
-$(BUILD)/firmware.elf: $(OBJ_COMMON) $(OBJ_RUN)
-	$(Q)$(LD) $(LDFLAGS) -o $@ $^ $(LIBS)
+$(BUILD)/firmware.elf: $(LDSCRIPT) $(OBJ_COMMON) $(OBJ_RUN)
+	$(Q)$(LD) $(LDFLAGS) -o $@ -T $^ $(LIBS)
```
Comment (dpgeorge) [inline ports/qemu-arm/Makefile]: This is a bit unconventional/confusing: the `-T` is consuming the first part of `$^`, with the rest of `$^` being the object files.  I think it'd be cleaner (less confusing) if there were a new variable like `OBJ = $(OBJ_COMMON) $(OBJ_RUN)` that was used in place of `$^`, and `$(LDFLAGS)` left as-is.

Comment (Jongy) [reply ports/qemu-arm/Makefile]: Right

Comment (Jongy) [reply ports/qemu-arm/Makefile]: Fixed

**Comment by dpgeorge** [comment]
Thanks for updating.  Merged in 973f68780d2eb974e4921ffdf513079efc19e0a4

**Pull request closed without merging**